### ISSUE TITLE
Dependency conflict for rxjava with micronaut BOM

### ISSUE THREAD

**Issue by esocode** [micronaut-projects/micronaut-data#415]
When trying to inject a bean with the RxJava type `Single` I just got the error message _Multiple possible bean candidates found: [io.reactivex.Single, io.reactivex.Single]_. I then noticed that there are two versions of RxJava2 on the classpath: 2.2.10 from the micronaut BOM and 2.2.17 from  `micronaut-data/data-model/build.gradle`. As micronaut-data is referenced from the BOM shouldn't it use the dependencies defined therein?

### Environment Information

- **Micronaut Version:** 1.3.2

**Comment by graemerocher** [comment]
the better question is why you are trying to inject a `Single` object?

**Comment by esocode** [comment]
Yeah, good question :-) That was just an experiment, I had changed the code soon after. But I thought that this dependency conflict could cause other problems too, therefore reported it.

**Comment by esocode** [comment]
Although, to elaborate a bit further, what I was trying to achieve was the following: I need to fetch a bunch of data from a REST service with a reactive API, i.e. returning a `Single`. I need to feed this data into legacy code which doesn't handle reactive classes, so I have to blockingGet it first. 

First I tried to do that in the post-construction code of a factory that provides the data at the injection points, but that lead to a timeout upon app start. My idea then was to return and inject the `Single` wrapping the data itself (as a Singleton bean) so that the request could continue in the background and the code invoking the legacy code would do the blockingGet on demand. But that lead to the error described above. I then moved the blockingGet to factory methods and now it works but wouldn't the use case with injecting a `Single` be valid for such background initializations?
